src/content/reference/device-os/firmware.md

```diff
@@ -12979,6 +13005,1754 @@ int8_t measurePower() const;
+
+
+### toArray() [Variant class]
+
+{{api name1="Variant::toArray()"}}
+
+Returns the value of the variant as an array.
+
+| Source | Result | 
+| :--- | :--- |
+| VariantArray | unchanged |
+| other | conversion fails |
+
+This method takes an `ok` parameter passed by reference, which is filled in as 
+`true` if the data is already this type.
+
+```cpp
+// PROTOTYPES
+VariantArray toArray(bool& ok) const;
+```
+
+### asArray() [Variant class]
+
+{{api name1="Variant::asArray()"}}
+
+Returns a reference to the value contained in this variant as an `VariantArray`. This can be used to modify the value of the variant and to more efficiently read the array without having to copy it.
+
+```cpp
+// PROTOTYPE
+VariantArray& asArray();
+```
+
+This method will not convert the type of variant. If the Variant is not already a `VariantArray` then this method will assert.
+
+
+### toMap() [Variant class]
+
+{{api name1="Variant::toMap()"}}
+
+Returns the value of the variant as an map.
+
+| Source | Result | 
+| :--- | :--- |
+| VariantMap | unchanged |
+| other | conversion fails |
+
+This method takes an `ok` parameter passed by reference, which is filled in as 
+`true` if the data is already this type.
+
+```cpp
+// PROTOTYPES
+VariantMap toMap(bool& ok) const;
+```
+
+
+### asMap() [Variant class]
+
+{{api name1="Variant::asMap()"}}
+
+Returns a reference to the value contained in this variant as an `VariantMap`. This can be used to modify the value of the variant and to more efficiently read the map without having to copy it.
+
+```cpp
+// PROTOTYPE
+VariantMap& asMap();
+```
+
+This method will not convert the type of variant. If the Variant is not already a `VariantMap` then this method will assert.
+
+### toJSON [Variant class]
+
+{{api name1="Variant::toJSON()"}}
+
+Serializes a VariantArray or VariantMap as JSON and returns it in a `String` object.
+
+```cpp
+// PROTOTYPE
+String toJSON() const;
+```
+
+### fromJSON [Variant class]
+
+{{api name1="Variant::fromJSON()"}}
+
+Given a JSON object as a c-string or `JSONValue`, returns a `Variant` object.
+
+```cpp
+// PROTOTYPE
+static Variant fromJSON(const char* json);
+static Variant fromJSON(const JSONValue& val);
+
+// EXAMPLE
+const char *jsonStr = "{\"a\":123}";
+Variant data = Variant::fromJson(jsonStr);
+```
+
+## VariantArray
+
+{{api name1="VariantArray"}}
+
+The `VariantArray` is a [Vector](#vector), essentially a dynamically-sized array of Variants, which are containers for arbitrary data. Think of this as a container to hold arbitrary data in Variant objects before converting to a JSON or CBOR array before sending across the network.
+
+```
+// DEFINITION
+typedef Vector<Variant> VariantArray;
+```
+
+The methods below are implemented in the `Variant` class, so you can use them update a `Variant` that is of type `VariantArray`.
```

Comment:
> you can use them update a `Variant` that is of type `VariantArray`

The array methods defined in `Variant` will convert the variant to an array so it's not required for the variant to be an array prior to the operation. The map methods will convert the variant to a map.

Reply:
Fixed